source/service.go

```diff
@@ -33,14 +36,16 @@ type serviceSource struct {
 	namespace string
 	// set to true to process Services with legacy annotations
 	compatibility bool
+	fqdntemplate  string
 }
 
 // NewServiceSource creates a new serviceSource with the given client and namespace scope.
-func NewServiceSource(client kubernetes.Interface, namespace string, compatibility bool) Source {
+func NewServiceSource(client kubernetes.Interface, namespace string, compatibility bool, fqdntemplate string) Source {
 	return &serviceSource{
 		client:        client,
 		namespace:     namespace,
 		compatibility: compatibility,
+		fqdntemplate:  fqdntemplate,
```

Comment:
template object should not be generated on each call. I would prefer to have the template object created in NewServiceSource() with possible returned error, and having `tmpl` as the object field .